src/table.ts

```diff
@@ -16,38 +16,37 @@
 
 'use strict';
 
-import {promisifyAll} from '@google-cloud/promisify';
+import { promisifyAll } from '@google-cloud/promisify';
 import * as is from 'is';
-import {ServiceError} from 'grpc';
+import { ServiceError } from 'grpc';
 import * as through from 'through2';
+import { Operation as GaxOperation } from 'google-gax/build/src/longrunning';
 
-import {Json} from './codec';
-import {Database} from './database';
-import {DatabaseAdminClient as d, SpannerClient as s} from './v1';
-import {PartialResultStream, Row} from './partial-result-stream';
-import {Snapshot, ReadRequest, Transaction, TimestampBounds} from './transaction';
+import { google as spanner_client } from '../proto/spanner';
+import { google as database_admin_client } from '../proto/spanner_database_admin';
 
-export type Key = string|string[];
+import { Json } from './codec';
+import { Database } from './database';
+import { PartialResultStream, Row } from './partial-result-stream';
+import { ReadRequest, TimestampBounds, Transaction, Snapshot } from './transaction';
+import { Schema } from './common';
 
-type Schema = string|string[]|{statements: string[], operationId?: string};
+export type Key = string | string[];
 
-type CommitPromise = Promise<[s.CommitResponse]>;
-type CreateTablePromise = Promise<[Table, d.Operation, d.GrpcOperation]>;
-type DropTablePromise = Promise<[d.Operation, d.GrpcOperation]>;
-type ReadPromise = Promise<[Array<Row|Json>]>;
-
-interface CreateTableCallback {
+export interface CreateTableCallback {
   (err: ServiceError, table?: null, operation?: null, apiResponse?: null): void;
-  (err: null, table: Table, operation: d.Operation,
-   apiResponse: d.GrpcOperation): void;
+  (err: null, table: Table, operation: GaxOperation,
+    apiResponse: DeleteResponse): void;
 }
-
-interface DropTableCallback {
+export type CreateTableResponse = [Table, GaxOperation, database_admin_client.longrunning.Operation];
+export type CommitResponse = [spanner_client.spanner.v1.CommitResponse];
+export type CommitCallback = spanner_client.spanner.v1.Spanner.CommitCallback;
```

Comment:
We can't use the generated callbacks, ever, because they return the grpc message class and we want the interface counterpart.

Reply:
That's going to be fun with the rest of these PR's 🤦‍♂️ gotcha, purely manual then

Reply:
Yeah, unfortunately :( if we ever get typescript gapics then we should be able to use those, but in the mean time I would just make a generic to try and simplify the types.

```typescript
interface RequestCallback<T> {
  (err: ServiceError|null, response?: T|null): void;
}

export type CommitCallback = RequestCallback<google.spanner.v1.ICommitResponse>;
```

Reply:
Good thinking - okay, I'll do that